### ISSUE TITLE
Allow only while using the app in android 10 returns GeolocationStatus.denied

### ISSUE THREAD

**Issue by humazed** [Baseflow/flutter-geolocator#334]
## 🐛 Bug Report

when selecting Allow only while using the app the new option in Android 10  `Geolocator().checkGeolocationPermissionStatus()` returns  `GeolocationStatus.denied`
![Screenshot_1570103723](https://user-images.githubusercontent.com/8838182/66124725-75509e80-e5e5-11e9-899d-630affe6878e.png)

**Version:** 5.1.4

**Platform:** 
:robot: Android

**Comment by StijnWoerkom** [comment]
Could you tell me more about the code that returns GeolocationStatus.denied?

**Comment by humazed** [comment]
I can reproduce it with the sample app.

**Comment by StijnWoerkom** [comment]
Does the emulator solve the GeolocationStatus.denied error?

**Comment by humazed** [comment]
I'm not sure I understand what you mean

**Comment by StijnWoerkom** [comment]
Could you tell me some details about the code and what you are trying to create? I am trying to reproduce your issue, but I need more information.

**Comment by humazed** [comment]
try the demo app on an android emulator with android 10
![untitled](https://user-images.githubusercontent.com/8838182/66307264-477c9a00-e904-11e9-8d79-3df9c332fce5.gif)

**Comment by StijnWoerkom** [comment]
I reproduced this issue and came to the conclusion that the location settings from the device were turned off. To fix this issue go to **Settings**>**Location**>**Use Location**>**On**.

**Comment by humazed** [comment]
the location was turned on I checked it, hence when I choose **allow all the time** it recognized as granted.

![untitled](https://user-images.githubusercontent.com/8838182/66384911-e0bcb680-e9bf-11e9-9d2d-86f0fd50f8ba.gif)

**Comment by StijnWoerkom** [comment]
The output of the app says you are using FusedLocationProvider.  It's recommended to use Google Play services version 11.6.0 or higher, which includes bug fixes for this class.

**Comment by gkrawiec** [comment]
Its a completely new permission in the new Android 10. it should neither give Allowed or Denied since it requires different checks.

https://developer.android.com/reference/android/Manifest.permission.html#ACCESS_BACKGROUND_LOCATION

**Comment by StijnWoerkom** [comment]
@humazed  What are you trying to achieve with the Geolocator().checkGeolocationPermissionStatus() method? I don't get any errors if I set the location settings to Allowed all the time.

**Comment by StijnWoerkom** [comment]
I reproduced the demo app in the new Android 10 and with the Geolocator().checkGeolocationPermissionStatus() method and retrieved 0 errors when I choose to allow only while using the app. Could you send your code so I could take a look at it?

**Comment by humazed** [comment]
I just cloned the repo and ran it without any changes

**Comment by humazed** [comment]
I even created a new emulator in case my current one has a problem or something but getting the same behavior

**Comment by StijnWoerkom** [comment]
I think the issue would depend more on the settings from the emulator. The app states that there are issues with Google Play Services. Are you using a network or are you at a location that does not support the Google Play Services?

**Comment by humazed** [comment]
I have tried with a fresh emulator but the problem is the same 

The app works fine for any version lower than android10 

In the gifs I provided I'm showing the version of Google play services and the the location is enabled

**Comment by PlymouthRock** [comment]
I have this same issue if using version 5.1.4+1 and compiling with SDK 28.  If I compile on SDK 29 it works.  Running this on a Pixel 3a running Android 10.

**Comment by adithyaxx** [comment]
> I have this same issue if using version 5.1.4+1 and compiling with SDK 28. If I compile on SDK 29 it works. Running this on a Pixel 3a running Android 10.

I can confirm that setting `compileSdkVersion` and `targetSdkVersion` to 29 resolved this issue.

**Comment by tampiss** [comment]
I have the same issue.

**Comment by nachiketkhasbag** [comment]
I see that with iOS too. getCurrentPosition() shows dialog box to enable location always or enable location while using the app or deny location access. Selecting "Enable while using the app" results in an exception with an error "PERMISISON_DENIED"

I am using geolocation 5.1.3

**Comment by illiashvedov** [comment]
A similar error is reproduced on Pixel 4.

**Comment by DenisBogatirov** [comment]
Same issue.
Samsung Galaxy S10 Plus, when click on "Keep while-in-use access" get PermissionStatus.denied

**Comment by StijnWoerkom** [comment]
We are currently working on a new version of the location permissions plugin. This new version includes respected while-in-use access and no longer returns PermissionStatus.denied.

**Comment by StijnWoerkom** [comment]
https://github.com/Baseflow/flutter-permission-plugins/issues/21

**Comment by mulderpf** [comment]
Urgh - I just discovered that I am experiencing the same issue for a production app.  I had wrongly assumed that when a user gives in-use access permissions, this method would return PermissionStatus.granted - so I can no longer rely on this it seems.

**Comment by JBBx2016** [comment]
"I can confirm that setting compileSdkVersion and targetSdkVersion to 29 resolved this issue." solved the issue until I added this in the AndroidManifest.xml

`<uses-permission android:name="android.permission.ACCESS_BACKGROUND_LOCATION"/>`

**Comment by JBBx2016** [comment]
By making location_permissions respect LocationPermissionLevel in the checkPermissionStatus method, I prevented Geolocator.getCurrentLocation from calling requestPermissions after the user has granted when in use permission. 

https://github.com/JBBx2016/flutter-permission-plugins/commit/29a1124cf805f549112a3534970fc762ca61a6c2

**Comment by StijnWoerkom** [comment]
@JBBx2016 We are working on the android.permissions.ACCES_BACKGROUND_LOCATION permissions. When the permissions are fixed, it could solve your error.

**Comment by mustakimpawle** [comment]
Any update?

**Comment by andreystavitsky** [comment]
any update?

**Comment by lkyashu** [comment]
Any Update

**Comment by praveenb** [comment]
Not working on Android 10 devices. :|

**Comment by andreystavitsky** [comment]
As a temporary workaround, you can use my fork
[branch Master](https://github.com/andreystavitsky/flutter-geolocator/commits/master)
where I moved permission stuff from obviously broken location_permissions to another plugin permission_handler

**Comment by louia** [comment]
My app just stop working after Google Play service PopUp, what ever i answer "Deny" , "Authorize", etc.
Android 10 with emulator.

**Comment by David-GIC** [comment]
still don't have solution zzzzz

**Comment by it-sam** [comment]
```
Geolocator locator = Geolocator();
    locator
        .checkGeolocationPermissionStatus(
            locationPermission: GeolocationPermission.locationWhenInUse)
```
**Need to pass permission in params.**
GeolocationPermission.locationWhenInUse

**Comment by mvanbeusekom** [comment]
Fixed in version 6.0.0
